Signal/src/util/UIViewController+Permissions.m

```diff
@@ -0,0 +1,101 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "Signal-Swift.h"
+#import "UIUtil.h"
+#import "UIViewController+Permissions.h"
+#import <AVFoundation/AVFoundation.h>
+#import <SignalServiceKit/Threading.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation UIViewController (Permissions)
+
+- (void)ows_askForCameraPermissions:(void (^)(BOOL granted))callbackParam
+{
+    DDLogVerbose(@"[%@] ows_askForCameraPermissions", NSStringFromClass(self.class));
+
+    // Ensure callback is invoked on main thread.
+    void (^callback)(BOOL) = ^(BOOL granted) {
+        DispatchMainThreadSafe(^{
+            callbackParam(granted);
+        });
+    };
+
+    if ([UIApplication sharedApplication].applicationState != UIApplicationStateActive) {
```

Comment:
Never ask for camera or microphone permissions if app is not active. I think I'm going to modify this to only skip if in background. iOS could temporarily make app inactive to show a system alert, e.g. "iOS update available."